データベースの設計

yabecchi312 opened the pull request:
#WHAT
データベースを設計する
#HOW
サービスに必須の機能であるため

Commit:
create pull request

Commit:
create users table

usersテーブルを作成しました。

Commit:
create members table

groupsテーブルとusersテーブルの中間テーブルmembersテーブルを作成しました。

Commit:
create messages table

messagesテーブルを作成しました。

Commit:
Add association

アソシエーションを追加しました。

Commit:
alter readme

readmeのmessagesテーブルの部分を修正。

Commit:
alter readme

readmeの修正

Commit:
alter readme

MrUrakou commented on README.md:
```diff
@@ -1,24 +1,51 @@
+|Column|Type|Options|
+|------|----|-------|
+|name|string|null: false, add_index|
+|email|integer|null: false, unique: true, add_index|
```
【質問】add_indexつける理由は何ですか？

yabecchi312 replied:
テーブルの検索速度を向上するためです。記述をindex: trueに変えて、更新速度が遅くなるので、userテーブルの方のindexは削除しました。

MrUrakou commented on README.md:
```diff
@@ -1,24 +1,51 @@
 
-* Ruby version
+### Association
+-  has_many :groups, through: :group_users
```
has_many :group_usersを先に記述しましょう。
コードは上から読まれるので、そうしないとアソシエーションがうまくいきません。

MrUrakou reviewed (commented):
【WIP】は実装途中で見て欲しい時に付けるので、今回は外しましょう。また、こちらはデータベースの設計のbranchなので、モデル作成の必要はありません（設計が終わってないのにモデル作成出来ないので。）

Commit:
alter association and index

・usersテーブルとgroupsテーブルのgroup_usersをmembersに変えて、associationのhas_many:membe
rsを先に記述。
・add_indexをindex:trueにしてusersテーブルのindexは削除

TOMARUMARU commented on README.md:
```diff
@@ -1,24 +1,52 @@
-* Configuration
+|Column|Type|Options|
+|------|----|-------|
+|group_name|string|null: false, index: true|
```
nameで大丈夫です！

TOMARUMARU commented on README.md:
```diff
@@ -1,24 +1,52 @@
+-  has_many :users, through: :members
 
-* Database initialization
+-  accepts_nested_attributes_for :members
```
accepts_nested_attributesはchatspaceでは使用しません！

TOMARUMARU commented on README.md:
```diff
@@ -1,24 +1,52 @@
+
+|Column|Type|Options|
+|------|----|-------|
+|body|text|null: false|
```
bodyかimageどちらかで投稿できますので、null falseは外しましょう！

TOMARUMARU commented on README.md:
```diff
@@ -1,24 +1,52 @@
+|Column|Type|Options|
+|------|----|-------|
+|body|text|null: false|
+|image|string|null: false|
```
bodyかimageどちらかで投稿できますので、null falseは外しましょう！

TOMARUMARU reviewed (commented):
上でも指摘されていますが、このブランチでモデル作成するのはやめましょう。
そもそもデータベース設計が終わっていないのにモデルが作成されているのはおかしいです。

yabecchi312 commented:
わかりました。気をつけます。

Commit:
alter detabase

group_name→name
messagesのimageとbodyのnullを外す。

TOMARUMARU commented on README.md:
```diff
@@ -1,24 +1,50 @@
-* Configuration
+|Column|Type|Options|
+|------|----|-------|
+|name|string|null: false, index: true|
```
group名はindexいらないと思います！

TOMARUMARU reviewed (commented):
rollbackをしてテーブルおよびモデルを削除してください！

Commit:
alter readme

データベースとモデルを削除。
readmeの修正。

yabecchi312 commented:
すみません、テーブルとモデルの削除に手間取ってしまいましたが削除完了しました。

TOMARUMARU reviewed (commented):
schmaにマイグレーションファイルの変更履歴が残ってしまっています。。。
このままだと次にモデルを作成するときにエラーが出てしまいます。

Commit:
destroy model

modelの削除

yabecchi312 commented:
度々すみません、schmaの中身のmodelがおそらく削除できたかと思います。
ご確認お願い致します。

ghost commented:
#の後に半角スペースを入れるとマークダウンを反映することができます！
こんな感じです！
今回は大丈夫ですが、次回から気をつけましょう！
# What
# Why

ghost commented:
LGTMです〜！
